### DISCUSSION TITLE
Type inference issue when state defined as non-arrow function

### DISCUSSION THREAD

**Discussion by tslocke** [vuejs/pinia#640] [Help and Questions] [answered]
The second of these two has a type error. The only difference is the arrow fn for state.

```
const useGood = defineStore('good', {
  state: () => ({
    counter: 0
  }),
  getters: {
    doubleCount: (state) => state.counter * 2,
  },
})

const useBad = defineStore('bad', {
  state() {
    return { counter: 0 }
  },
  getters: {
    doubleCount: (state) => state.counter * 2 // Property 'counter' does not exist on type '{} & {}'
  },
})
```

Is this expected? If so would maybe the docs should always use an arrow fn for state? This tripped me up just now. I'd be happy to submit a doc PR.

**Comment by posva** [comment, marked as answer]
I'm not sure of why the type isn't inferred.

> Is this expected? If so would maybe the docs should always use an arrow fn for state? This tripped me up just now. I'd be happy to submit a doc PR.

Yes, let's use only arrow functions for state in docs.
